### ISSUE TITLE
Question: v17 SPMC differences

### ISSUE THREAD

**Issue by Zxurian** [koying/SPMC#754]
I'll preface this by saying I've been using v16 SPMC on my Shield & FireStick and they were great. Where v16 always failed when trying to do passthrough on android, SPMC was able to do it fine.

With the advent of v17, I tried it last night and it looks like the audio issues were taken care of, I'm able to passthrough without an issue.

One of the other major features I used was the easy ability to add MySQL information via the Kodi GUI, instead of having to go through ridiculous file managers or SSH to edit the advancedsettings.xml.

Is there any general list on what the differences will be in the SPMC v17 vs Kodi v17? I've been holding off on upgrading my libraries since my android devices are all using SPMC v16, and I'm ok with waiting for SPMC v17 if it's got a bunch of Android improvements coming down the pike.

As a side note, is there any particular reason why official Kodi doesn't _want_ to include a MySQL GUI editor under system settings?

**Comment by JFG90** [comment]
Is SPMC 17 available to test?

**Comment by JFG90** [comment]
Sorry I'm getting confused from reading it I thought you had tested SPMC 17 lol

**Comment by Zacharybinx34** [comment]
I would love to get v.17 rolling on spmc here.

**Comment by 2WhlWzrd** [comment]
Am also patiently awaiting word on SPMC 17.

**Comment by Theetjuh** [comment]
Yeah would love v17 aswell, so I can upgrade my other devices to v17 as well using a MySQL DB.
They already got updates waiting, so posponing those untill SPMC is at Krypton :)

Am really looking forward to the new music section and ofc having all devices on the latest version is handy.

**Comment by ghost** [comment]
Patience Patience!

**Comment by koying** [comment]
Guys. Read the Kodi Android forum and count the number of regressions mentioned, ie "it worked on 16.1 and SPMC".

Surely, Krypton is not ready for everyday use. Maybe 17.1 or 17.2...

**Comment by Zacharybinx34** [comment]
Who cares. I'm sure they'll get fixed quickly. v. 16 is dead and many kodi ui's no longer support it.

**Comment by koying** [comment]
I surely care more for a working Kodi/SPMC than "kodi ui's no longer support it"...

**Comment by Zacharybinx34** [comment]
That's fair. I'm just really looking forward to the new v.17 features, like the re-done dvd player. Hoping it helps with stability a lot. 

Any clue when you think you'll make the switch? Or is that just 100% dependent on when v.17's bugs are gone?

**Comment by ghost** [comment]
This 

> Krypton is not ready for everyday use. Maybe 17.1 or 17.2

Best to contemplate & wait!

**Comment by Zxurian** [comment]
@koying I agree. I'd rather wait for a more stable version to start from, vs a broken foundation. I was more wondering what your plans for SPMC v17 would be in terms of changes from Kodi. Some of the things that SPMC v16 did that Kodi didn't seem to have been rolled in (passthrough for example, at least on the Shield). Are there any major features you're planning that you see Kodi v17 didn't include?

**Comment by 99rook99** [comment]
Personally, I consider Kodi 17, to be a transitional release. With the number of changes to the core code, I'm looking more towards 18 than 17.

**Comment by s2law** [comment]
I've been using 17 for a while and it's been excellent, just as stable as Kodi 16 and SPMC, but faster than both! Certainly from my experience anyway. It's a shame that SPMC isn't making the transition yet, but it's your baby and we'll wait for it (bregrudgingly)

**Comment by hydraSlav** [comment]
There is just one singular tiny feature that I need from Krypton: context menu on homescreen widgets. Is there any way to port over just that feature into SPMC v16, while we wait for the rest of Krypton to stabilize?

**Comment by lllopo** [comment]
I gave a try to Krypton on my Shield and it runs just fine with everything I threw at it. More importantly - it runs smoother than both Jarvis and SMPC. Actually, I've had more issues (and crashes here and there) with SPMC than Krypton. The thing is, I realy miss the excellent SPMC integration with the Shield, so I"m also looking forward for the Kryptonized SPMC.

**Comment by Zacharybinx34** [comment]
Yeah, same experience. Kryp;ton seems to have solved a lot of the instability that plagued Kodi in the past. 

Hopefully we get Krypton + SPMC soon!

**Comment by dm2912** [comment]
Krypton my firestick (tried on both) was a painful experienced. both upgrades, and fresh installs. primary problem seemed to be connectivity in that it was taking maybe 5 minutes to load up menus/folders. was like someone had stolen my fibre and replaced it with diallup. actually interface speed was fine and zippy, just anything that involved remote folders was painfully slow. but with the various other bugs i honestly dont think its useable (on some devices) as a everyday version

**Comment by ghost** [comment]
That's why @koying specified wait for at least versions 17.1 or 17.2 before jumpin' in the fire!

**Comment by charettepa** [comment]
the new video player in 17 is ok in windows but horrible on android.  if you watch live tv "mpeg/.ts" it constantly stutters and crashes. 

but SPMC based on 16 is rock solid with no issues.

17 is NOT ready for primetime

**Comment by lllopo** [comment]
By coincidence - I just tried a 4K .ts on my Shield last night on SPMC/16 - didn't started at all, just "please wait ...". Haven't tried it on 17, cause I've put 17 just to take a look and removed it last week. 
My point is - user experiences varie a lot and you can not draw conclusins that 17 is not ready at all and you see - 16 is a rock ... it is simply not like this. As I said before - everything 1080p that I threw at the 17 on my Shield just played flawlessly, not a single crash and snappier than 16 ... so it is different for each user and each device and in my view it is good to see an SPMC/17 release. Besides, once done,  merging an eventual 17.1 or 17.2 wouldn't be such a big hassle as an initial release, I believe. 
Ow, and btw., the 4K .ts played ok in Plex ... go figure.

**Comment by charettepa** [comment]
please remember that shield is much more powerful than the nexus player
i have both

in kodi 17 hw acceleration is not working properly
this does not affect shield as its powerful enough but the nexus player
suffers from it

**Comment by lllopo** [comment]
@charettepa That's exactly my point - the device diversity is big and 17 does not work for you just as much as it's ok for me and vice versa - 16 is perfect for you, but it is not-so-good for me. So this is what I"m saying - this shouldn't be a criteria for when do we migrate to 17, since as soon as it is avaiable - it becomes a matter of personal choice. 
Now, on other hand - if it is too much of work to merge SPMC code to each 17 iteration (17.1, 17.2 etc.), then I'll totally agree with you (and @koying for that matter), but I don' think this is the case (a wild guess from my side, of course).

**Comment by TommyGunn69** [comment]
17.1 is now out
